### ISSUE TITLE
Azure blob storage prepends a '/' to the blob names when the prefix is None

### ISSUE THREAD

**Issue by shikharsg** [zarr-developers/zarr-python#525]
Azure blob storage prepends a '/' to the blob names, even when the prefix is None. Code sample:

#### Minimal, reproducible code sample
This code needs to connect to Azure Blob (the sample below connects to a local emulator)
```python
>>> import zarr
>>> store = zarr.ABSStore('test', prefix=None, blob_service_kwargs={'is_emulated':True})
>>> g = zarr.open_group(store)
>>> ## list all blobs in the container
>>> print([blob.name for blob in store.client.list_blobs('test')])
['/.zgroup']
```
which shows up in the storage explorer like this:
![image](https://user-images.githubusercontent.com/8643775/70611383-03a45a80-1bfd-11ea-92ea-cc6c94330ca0.png)

There should ideally be no '/' before the `.zgroup` and it should look like this:
```python
>>> import zarr
>>> store = zarr.ABSStore('test', prefix=None, blob_service_kwargs={'is_emulated':True})
>>> g = zarr.open_group(store)
>>> ## list all blobs in the container
>>> print([blob.name for blob in store.client.list_blobs('test')])
['.zgroup']
```
![image](https://user-images.githubusercontent.com/8643775/70611581-541bb800-1bfd-11ea-97dd-c3eb9f62440f.png)

Will submit a PR for this in a bit

**Comment by tjcrone** [comment]
It's not clear to me why you would want .zgroup to be directly under a blob container. The blob container is not a Zarr group.

**Comment by shikharsg** [comment]
Fixed in PR #526
